src/server/pfs/s3/server.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gogo/protobuf/types"
+
+	"github.com/pachyderm/pachyderm/src/client"
+)
+
+const locationResponse = `<?xml version="1.0" encoding="UTF-8"?>
+<LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/">PACHYDERM</LocationConstraint>`
+
+type handler struct {
+	pc *client.APIClient
+}
+
+func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	parts := strings.SplitN(r.URL.Path, "/", 4)
+
+	if len(parts) == 2 && parts[1] == "_ping" {
+		// matches `/_ping`
+		w.Write([]byte("OK"))
```

Comment:
Responding to `ping` with anything but `pong` seems borderline illegal. Not an actual request for change here.